avr-hal-generic/src/i2c.rs

```diff
@@ -499,5 +499,82 @@ macro_rules! impl_twi_i2c {
                 Ok(())
             }
         }
+
+        $(#[$i2c_attr])*
+        pub struct Slave<CLOCK: $crate::clock::Clock, M> {
```

Comment:
The previous suggestion of `$I2cSlave` for a name was causing conflicts, so I used `Slave` temporarily...